pulp_container/app/models.py

```diff
@@ -503,6 +502,7 @@ def validate(self):
             )
 
             with open(signed["signature_path"], "rb") as fp:
+                gpg = gnupg.GPG()
```

Comment:
Hmm, this is not a full solution to the problem. Yes it lets the sevices start just to fail later when the actual verification is attempted. But i think it is still a good change.

Reply:
I don't think we should do anything else.
I think the problem was in the PATH not being set on the user side
```
In [18]: cmd = ['gpg', '--version']

In [19]: result = Popen(cmd, shell=False, env={'PATH': ''})
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-19-f2e3bdace1dd> in <module>
----> 1 result = Popen(cmd, shell=False, env={'PATH': ''})

/usr/lib64/python3.9/subprocess.py in __init__(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, user, group, extra_groups, encoding, errors, text, umask)
    949                             encoding=encoding, errors=errors)
    950 
--> 951             self._execute_child(args, executable, preexec_fn, close_fds,
    952                                 pass_fds, cwd, env,
    953                                 startupinfo, creationflags, shell,

/usr/lib64/python3.9/subprocess.py in _execute_child(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, gid, gids, uid, umask, start_new_session)
   1819                     if errno_num != 0:
   1820                         err_msg = os.strerror(errno_num)
-> 1821                     raise child_exception_type(errno_num, err_msg, err_filename)
   1822                 raise child_exception_type(err_msg)
   1823 

FileNotFoundError: [Errno 2] No such file or directory: 'gpg'

In [20]: cmd = ['/bin/gpg', '--version']

In [21]: result = Popen(cmd, shell=False, env={'PATH': ''})

gpg (GnuPG) 2.2.27
libgcrypt 1.9.2-unknown
Copyright (C) 2021 Free Software Foundation, Inc.
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/vagrant/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

Reply:
I'm unsure if it's considered good practice to set the PATH environment for daemons though. We should ask an experienced admin. @mikedep333 ?

Reply:
https://github.com/pulp/pulp_installer/blob/main/roles/pulp_api/templates/pulpcore-api.service.j2#L16
I've asked the user to share the systemd config file for pulpcore-api worker to see what PATH is set there.

Reply:
@ipanova Correct, it's invalid syntax. Now the question is whether we (pulp_installer) accidentally put $PATH in it, or whether the user did.